x/msg_authorization/internal/types/msgs.go

```diff
@@ -0,0 +1,124 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"time"
+)
+
+// MsgGrant grants the provided capability to the grantee on the granter's
+// account with the provided expiration time.
+type MsgGrantAuthorization struct {
+	Granter    sdk.AccAddress `json:"granter"`
+	Grantee    sdk.AccAddress `json:"grantee"`
+	Capability Capability     `json:"capability"`
+	// Expiration specifies the expiration time of the grant
+	Expiration time.Time `json:"expiration"`
+}
+
+func NewMsgGrantAuthorization(granter sdk.AccAddress, grantee sdk.AccAddress, capability Capability, expiration time.Time) MsgGrantAuthorization {
+	return MsgGrantAuthorization{
+		Granter:    granter,
+		Grantee:    grantee,
+		Capability: capability,
+		Expiration: expiration,
+	}
+}
+
+func (msg MsgGrantAuthorization) Route() string { return RouterKey }
+func (msg MsgGrantAuthorization) Type() string  { return "grant_authorization" }
+
+func (msg MsgGrantAuthorization) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Granter}
+}
+
+func (msg MsgGrantAuthorization) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
```

Comment:
Feedback from my prs on core, they prefer this as a one-liner:

`return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))`

Same functionality, different syntax. And they are pretty picky on syntax

Reply:
Sure, changed accordingly